frb_codegen/src/parser/mod.rs

```diff
@@ -16,9 +18,73 @@ use crate::method_utils::FunctionName;
 use crate::parser::ty::TypeParser;
 use crate::source_graph::Crate;
 
+use self::ty::convert_ident_str;
+
 const STREAM_SINK_IDENT: &str = "StreamSink";
 const RESULT_IDENT: &str = "Result";
 
+pub(crate) fn get_toposort(src: HashMap<String, Type>) -> HashMap<String, Type> {
```

Comment:
uber-nit: maybe we can rename it to be more clear what it is about? currently, the name sounds to me like "get a toposort object" and reads like a plain old getter (but it definitely is not)

Reply:
not sure the exact functionality of the function, so may suggest a name later after seeing unit tests etc

Reply:
@fzyzcjy  I have added unittest.

Reply:
I've tried so many times that I've lost track of how to change it.
Forgive me for my ability to resist frustration.

Also, because ret is a return value, I give it ownership. If you want it to be a reference, I think you have to reset the ownership.
And of course, if you look below,
You have ownership of the key and you have ownership of the value because they're all functionally resolved.
If it is reference here, then go to clone and all this ownership will be wasted

Reply:
The new name LGTM :)